deployment/app.py

```diff
@@ -0,0 +1,389 @@
+    return jsonify(messages)
+
+def serve_pil_image(pil_img):
+    img_io = StringIO()
+    pil_img.save(img_io, 'JPEG', quality=70)
+    img_io.seek(0)
+    return send_file(img_io, mimetype='image/jpeg')
+
+
+print( 
+'''
+    +----------------------------------------+
+    |                                        |
+    |      Turi Create Inference Engine      |
+    |                                        |
+    +----------------------------------------+
+    
+
+                                          $ 
+                                        $$  
+                                     $$$    
+          $$$$                 $$$$$$$      
+      $$$$$$$$$$          $$$$$$$$$$        
+       $$$$$$$$$$$     $$$$$$$$$$$$$$       
+          $$$$$$$$$$$$$$$$$$$$$$$$$$$       
+            $$$$$$$$$$$$$$$$$$$$$$$$$$      
+             $$$$$$$$$$$$$$$$$$$$$$$$$      
+              $$$$$$$$$$$$$$$$$$  $$$$$     
+              $$$$$$$$$$$$   $$$    $$$$    
+             $$$$$$$$$$      $$        $$   
+            $$$   $$$$       $$         $   
+          $$$    $$$        $           $$  
+       $$$     $$$        $$            $$  
+    $$$      $$                             
+           $$                               
+
+    Pass in model (and data if necessary) files.
+    This file will take care of the rest of the Magic!
+
+    Routes
+    ------
+
+    `/info`
+
+    Returns general metadata about the model shown below:
+
+        + type
+        + expected inputs
+
+    `/predict`
+
+    The API surface to the predict function for each model.
+
+'''
+)
+
+app = Flask(__name__)
+# Set up the command-line options
+default_host="0.0.0.0"
+default_port="5000"
+parser = argparse.ArgumentParser()
+
+parser.add_argument('model', type=str,
+                    help='The path to the model file')
+
+parser.add_argument('data', type=str, nargs='?',
+                    help='(optional) The path to the data file')
+
+parser.add_argument('--debug', dest='debug', action='store_true', 
+                    help='print debug messages to stderr')
+
+args = parser.parse_args()
+
+ALLOWED_IMAGE_EXTENSIONS = set(['png', 'jpg', 'jpeg', 'gif'])
+ALLOWED_AUDIO_EXTENSIONS = set(['wav', 'mp3'])
+
+def allowed_image_file(filename):
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_IMAGE_EXTENSIONS
+
+def allowed_audio_file(filename):
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_AUDIO_EXTENSIONS
+
+data = None
+if args.data != None:
+    data = tc.SFrame(args.data)
+
+model = tc.load_model(args.model)
+
+@app.route('/info')
+def model_info():
+    result = dict()
+    result["model"] = str(type(model))
+    for field in model._list_fields():
+        result[field] = model._get(field) if json_serializable(model._get(field)) else str(model._get(field))
+    return return_success(result, True)
+
+@app.route('/predict', methods=['GET', 'POST'])
+def model_predict():
```

Comment:
I'm having a really hard time understanding this function. It's too long, but more importantly there are too many levels of nesting. I think it will be difficult for other developers to make changes here without introducing bugs.

At a high level what you're trying to do here is map toolkits to result processing logic. I would recommend creating a separate function for each top level code block. 

Also rather than lots of complicated top level if/elif-statements, it's probably cleaner to do that mapping with a dictionary. 

To make it cleaner, I would recommend restructuring the top level function to look something like this:
```python
@app.route('/predict', methods=['GET', 'POST'])
def model_predict():
    TOOLKIT_TO_PROCESSING_FUNCTION = {
        tc.toolkits.image_classifier.image_classifier.ImageClassifier: processing1,
        tc.toolkits.style_transfer.style_transfer.StyleTransfer: processing2,
        ......
        ......
    }

    model_type = type(model)
    if model_type not in TOOLKIT_TO_PROCESSING_FUNCTION:
        return_success({"error": "Unimplemented Toolkit"}, False)

    result = TOOLKIT_TO_PROCESSING_FUNCTION[model_type]()
    return_success(result, True)
}